lib/in_memory_data_store.rb

```diff
@@ -0,0 +1,110 @@
+require 'modules/data_store'
+require 'date'
+require 'record_collection'
+require 'record_map_collection'
+require 'entry_collection'
+require 'item_collection'
+require 'entry'
+require 'item'
+require 'record'
+
+module RegistersClient
+  class InMemoryDataStore
+    include DataStore
+
+    def initialize(config_options)
+      @config_options = config_options
+
+      @data = {
+        records: { user: {}, system: {} },
+        entries: { user: [], system: [] },
+        items: {},
+        user_entry_number: 0,
+        system_entry_number: 0
+      }
+
+    end
+
+    def add_item(rsf_line)
```

Comment:
It would be good to make these in terms of Ruby objects like `RegistersClient::Item` and `RegistersClient::Entry` and not RSF lines. This is because we don't want the client to need to know how to parse RSF lines. It also allows the client library to change how it downloads the register and not affect the user, i.e. we might start using the JSON endpoints one day instead of RSF.

Reply:
Yep, makes sense. The same will apply for the `append_entry` method too.